share/functions/funced.fish

```diff
@@ -122,8 +130,50 @@ function funced --description 'Edit function definition'
                     continue
                 end
                 echo (_ "Cancelled function editing")
+            else if test -n "$writepath"
+                if not set -q _flag_save
+                    echo (_ "Warning: the file containing this function has not been saved. Changes will be lost when fish is closed.")
+                    set -l prompt (printf (_ 'Save function to %s\? [Y/n]') "$writepath")
+                    read -p "echo $prompt\  " response
+                    if test -z "$response"
+                        or contains $response {Y,y}{E,e,}{S,s,}
+                        set _flag_save 1
+                    else if not contains $response {N,n}{O,o,}
+                        echo "I don't understand '$response', assuming 'Yes'"
+                        set _flag_save 1
+                    end
+                end
+                if set -q _flag_save
+                    # try to write the file back
+                    # cp preserves existing permissions, though it might overwrite the owner
+                    if not cp $tmpname "$writepath" 2>&1
```

Comment:
I suggest using the happy path first - `if cp`, and then the failure case in the `else`.